manifests/host.pp

```diff
@@ -1,17 +1,13 @@
 # == Define: dhcp::host
 #
 define dhcp::host (
-  $ip,
-  $mac,
-  $options = {},
-  $comment='',
-  $ignored = false,
+  Stdlib::Compat::Ipv4 $ip,
+  String $mac,
```

Comment:
This could be a regex. I think ```^([0-9A-Fa-f]{1,2}:){5}([0-9A-Fa-f]{1,2})$``` should work (based on https://stackoverflow.com/questions/4260467/what-is-a-regular-expression-for-a-mac-address#4260512 but modified for dhcpd.conf). On the other hand it's not a regression in strictness.